frontend/app/forms/inputs/text-input-group.component.js

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import { useCss } from "kremling";
+import { DataContainerContext } from "../data-container.component";
+
+export default function TextInputGroup(props) {
+  const dataContext = React.useContext(DataContainerContext);
+  var inputs = Object.assign({}, ...props.inputs);
+  const [inputList, setInputList] = React.useState([inputs]);
+  const scope = useCss(css);
+
+  return (
+    <div className="web-form-input" onBlur={evt => handleBlur(evt)} {...scope}>
+      <label>{props.groupLabel}</label>
+      <div>
+        {inputList.map((input, idx) => (
+          <React.Fragment key={`${input}-${idx}`}>
+            {Object.entries(input).map(([key, val]) => (
+              <div key={`${key}-${idx}`} className="web-form-input text-input">
+                <label>{key}</label>
+                <input
+                  type="text"
+                  value={input[key]}
+                  onChange={evt => handleChange(evt, key, idx)}
+                />
+              </div>
+            ))}
+            {inputList.length > 1 && (
+              <button
+                className="web-form-input"
+                onClick={evt => handleRemoveGroup(evt, idx)}
+              >
+                Remove {props.buttonLabel}
+              </button>
+            )}
+          </React.Fragment>
+        ))}
+      </div>
+      <button className="web-form-input" onClick={evt => handleAddGroup(evt)}>
+        Add {props.buttonLabel}
+      </button>
+    </div>
+  );
+
+  function handleChange(evt, key, idx) {
+    const values = [...inputList];
+    values[idx][key] = evt.target.value;
+    setInputList(values);
+  }
+  function handleAddGroup(evt) {
+    evt.preventDefault();
+    const values = [...inputList];
+    values.push(inputs);
+    setInputList(values);
+  }
+  function handleRemoveGroup(evt, i) {
+    evt.preventDefault();
+    const values = [...inputList];
+    values.splice(i, 1);
```

Comment:
What you have here is fine, but here's another way that might be easier:
```suggestion
const values = inputList.filter((item, index) => index !== i)
```